compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -954,12 +957,44 @@ private void shadowSymbolScope(Location pos,
         for (BVarSymbol symbol : symbols) {
             // since the var decl is now within lambda, remove scope entry from encl env.
             env.scope.entries.remove(symbol.name);
+            env.enclPkg.globalVariableDependencies.values().forEach(d -> d.remove(symbol));
```

Comment:
Note: When `from Employee emp in employees` is defined at module-level, `emp` will initially (initial compiler phases prior to desugar) considered a global var. Once de-sugared, it'll get moved into a lambda. Hence, removing it from `globalVariableDependencies`.